src/osd/OSDCap.cc

```diff
@@ -245,6 +245,24 @@ void OSDCapGrant::expand_profile() const
     profile_grants.emplace_back(OSDCapMatch(profile.pool_namespace),
                                 OSDCapSpec(osd_rwxa_t(OSD_CAP_R | OSD_CAP_W)));
   }
+
+  if (profile.name == "rbd") {
+    // RBD read-write grant
+    profile_grants.emplace_back(OSDCapMatch("", "", "rbd_children"),
+                                OSDCapSpec(osd_rwxa_t(OSD_CAP_CLS_R)));
```

Comment:
Sounds good -- it would also be nice to have the monitor validate the OSDCap as well. Right now, I can run ```ceph auth get-or-create client.test mon 'allow r' osd 'fdsfdfa'``` and it doesn't complain, but it will barf if you provide an unreadable MonCap.